### PR TITLE
Properly handle pagination when GCS doesn't give us full results.

### PR DESCRIPTION
**Pull request by meisterT**

### CODE REVIEW

**Comment by meisterT** [comment]
cc @fweikert @philwo

**Comment by fweikert** [comment]
Fixes https://github.com/bazelbuild/bazelisk/issues/239

File: repositories/gcs.go [hunk 1]
```diff
@@ -63,27 +63,44 @@ func getVersionHistoryFromGCS() ([]string, error) {
...
-	content, err := httputil.ReadRemoteFile(url, "")
-	if err != nil {
-		return nil, false, fmt.Errorf("could not list GCS objects at %s: %v", url, err)
+		baseURL = fmt.Sprintf("%s&prefix=%s", baseURL, prefix)
 	}
 
-	var response GcsListResponse
-	if err := json.Unmarshal(content, &response); err != nil {
-		return nil, false, fmt.Errorf("could not parse GCS index JSON: %v", err)
+	var result GcsListResponse
```
Comment (fweikert) [inline repositories/gcs.go:71]: You could replace this with a []string and an int

**Review by fweikert — APPROVED** [review]

File: repositories/gcs.go (hunk 1 above)
Comment (meisterT) [reply repositories/gcs.go]: Done, also fixed a mistake. PTAL.

**Review by fweikert — APPROVED** [review]

**Comment by fweikert** [comment]
A unit test would be nice, too, but for now I would be happy enough if this has been tested manually.

**Comment by meisterT** [comment]
> A unit test would be nice, too, but for now I would be happy enough if this has been tested manually.

Yeah, I tested manually by adding `&maxResults=10` to the URL and observing the results.

**Pull request merged**